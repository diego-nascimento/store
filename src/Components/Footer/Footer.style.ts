import styled from 'styled-components'
import { styles } from '../../styles/styles'

export const FooterWrapper = styled.footer`
  width: 100vw;
  background-color: ${styles.componentsDest};
  display: flex;
  align-items: center;
  color: #ccc;
  flex-direction: column;

`

export const FooterContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr;
  grid-template-areas:"aside FooterMain Infos" "aside FooterMain Infos" ;
  padding: 30px 0px;

  @media(max-width: 1300px){
    margin: 0px 20px;
  }

  @media(max-width: 800px){
    display: flex;
    flex-direction: column;
    padding: 30px 20px;

    p{
      font-size: .7rem;
    }
  }
`

export const FooterAside = styled.aside`
  grid-area: aside;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 100%;

  h2{
    font-size: 1rem;
    letter-spacing: 1px;;
    text-transform: uppercase;
  }

  @media(max-width: 800px){
    margin-bottom: 20px;
  }
`

export const Content = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  padding-top: 20px;
`

export const Info = styled.div`
  margin-left: 20px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  h2{
    font-size: .9rem;
    margin: 0px;
  }

  p{
    font-size: .8rem;
    opacity: .7;
    padding: 0px;
    margin: 0px;
  }
`

export const RedesSociais = styled.div`
  padding-top: 20px;
  display: flex;
  flex-direction: row;

  a{
    margin-left: 10px;
    :first-child{
      margin-left: 0px;
    }
  }
`

export const SVGContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  
  padding: 5px;
  background-color: #ccc;
 
  border-radius: 50%;
  svg{
    font-size: 25px;
    color: ${styles.componentsDest};  
  }
`

export const FooterMain = styled.main`
  grid-area: FooterMain;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;

  h2{
    font-size: 1rem;
    letter-spacing: 1px;;
    text-transform: uppercase;
    margin-bottom: 20px;
  }
  
  a{
    color: #fff; 
    margin: 0px;
    padding: 0px;
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
    margin: 5px 0px;
  }

  p{
    opacity: .7;
    padding: 0px;
    margin: 0px;
  }

  @media(max-width: 800px){
    display: flex;
    flex-direction: column;
    padding: 20px 0px;
    align-items: flex-start;
    border-top: 1px solid #ccc;
  }
`

export const ContainerFooterMain = styled.div`

`
export const FooterRightSide = styled.aside`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-end;


  @media(max-width: 800px){
    display: flex;
    flex-direction: column;
    padding-top: 20px;
    align-items: flex-start;
    border-top: 1px solid #ccc;
  }
`

export const ContainerPagarme = styled.div`
  background-color: white;
  padding: 10px;
  position: relative;
  margin-bottom: 10px;

  p{
    color: #111;
    font-size: .55rem;
    position: absolute;
    top: 5px;
  }
  img{
    width: 130px;
  }
`

export const FooterBottom = styled.div`
  height: 60px;
  width: 100vw;
  display: flex;
  justify-content: center;
  background-color: ${styles.componentsDest};
`

export const ContainerFooterBoottom = styled.div`
  border-top: 1px solid #ccc;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  margin: 0px 20px;
`

export const ContainerBandeiras = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-end;
  padding: 5px;
  
  img{
    width: 140px;
    border: 3px solid white;
  }
`
