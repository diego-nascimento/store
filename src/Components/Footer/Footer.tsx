import React from 'react'
import { FooterWrapper, FooterContainer, FooterAside, FooterMain, Content, Info, RedesSociais, SVGContainer, ContainerFooterMain, FooterRightSide, ContainerPagarme, FooterBottom, ContainerBandeiras, ContainerFooterBoottom } from './Footer.style'
import Link from 'next/link'
import { AiOutlineMail, AiOutlineInstagram, AiOutlineWhatsApp, AiOutlinePhone } from 'react-icons/ai'

const Footer: React.FC = () => {
  return (
    <FooterWrapper>
      <FooterContainer className="Container">
        <FooterAside>
          <h2>Informações de Contato</h2>
          <Content>
            <AiOutlineInstagram />
            <Info>
              <h2>Instagram</h2>
              <p>@loveshopdaari</p>
            </Info>
          </Content>
          <Content>
            <AiOutlineMail />
            <Info>
              <h2>Email</h2>
              <p>libidopirauba@gmail.com</p>
            </Info>
          </Content>
          <Content>
            <AiOutlinePhone />
            <Info>
              <h2>Telefone</h2>
              <p>(32) 99974-5208</p>
            </Info>
          </Content>
          <RedesSociais>
            <Link href="https://www.instagram.com/loveshopdaari/">
              <a >
                <SVGContainer>
                  <AiOutlineInstagram />
                </SVGContainer>
              </a>
            </Link>
            <Link href="https://wa.me/message/IMPCTMLVS27FJ1">
              <a >
                <SVGContainer>
                  <AiOutlineWhatsApp />
                </SVGContainer>
              </a>
            </Link>
          </RedesSociais>
        </FooterAside>
        <FooterMain>
          <ContainerFooterMain>
            <h2>Links Rapidos</h2>
            <Link href="https://www.instagram.com/loveshopdaari/">
              <a>
                <p>Sobre nós</p>
              </a>
            </Link>
            <Link href="/entrega">
              <a>
                <p>Entrega</p>
              </a>
            </Link>
            {/* <Link href="politicas-de-trocas">
              <a>
                <p>Politicas de trocas e devoluções</p>
              </a>
            </Link> */}
            <Link href="/produtos">
              <a>
                <p>Produtos</p>
              </a>
            </Link>
          </ContainerFooterMain>
        </FooterMain>
        <FooterRightSide>
          <ContainerPagarme>
            <img src="/ssl.svg" alt="ssl certified" />
          </ContainerPagarme>
          <ContainerPagarme>
            <p>POWERED BY</p>
            <img src="/pagarme-logo.png" alt="pagarme" />
          </ContainerPagarme>
          <ContainerBandeiras>
            <img src="/bandeiras.png" alt="bandeiras aceitas" />
          </ContainerBandeiras>
        </FooterRightSide>
      </FooterContainer>
      <FooterBottom>
        <ContainerFooterBoottom className="Container">
          <span>2021  Diego Nascimento - Todos os Direitos Reservados</span>
        </ContainerFooterBoottom>
      </FooterBottom>
    </FooterWrapper>
  )
}
export default Footer
