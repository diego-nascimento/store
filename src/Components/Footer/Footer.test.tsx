import Enzyme, { shallow } from 'enzyme'
import React from 'react'
import Footer from './Footer'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('FooterComponent', () => {
  test('Should render with no errors', () => {
    const wrapper = shallow(<Footer />)
    expect(wrapper).toBeTruthy()
  })

  test('Should match snapshot', () => {
    const wrapper = shallow(<Footer />)
    expect(wrapper).toMatchSnapshot()
  })
})
