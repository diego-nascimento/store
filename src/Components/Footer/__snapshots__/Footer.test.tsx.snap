// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['FooterComponent Should match snapshot 1'] = 'ShallowWrapper {}'
